doc/content/msr/msfr/thermo_model.md

```diff
@@ -0,0 +1,118 @@
+
+It should be noted that this block creates AuxVariables of these inputs and ouputs defined here and can be manipulated and exported to an exodus file for viewing.
+
+#### Chemical Composition
+
+Next, the `ChemicalComposition` block is used to set up the Thermochimica calculation and is described in more detail [here](https://mooseframework.inl.gov/source/actions/ChemicalCompositionAction.html). First the modified MSTDB V2.0 with noble gas data file is specified, and the temperature, pressure, and mass unites as chosen.
+
+!listing msr/msfr/thermochemistry/thermo.i block=ChemicalComposition
+
+Next, the auxiliary variables of the temperature and pressure are specified as `tfuel_nod` and `pressure_nod` respetively. It should be noted that the element Auxiliary variables were already declared in the `GlobalParams` block. Lastly, a `reinitilization_type` is specified to use cached Thermochimica solutions to speed up calculations.
+
+#### Auxiliary Variables
+
+The `AuxiliaryVariables` block is used to set up any additional auxiliary variables that are needed for the calculation. Here we set up the temperature `tfuel` and pressure `pressure` of the fuel salt. These are `CONSTANT Monomials` corresponding to MOOSE Finite Volume variables which are initialized by the `initial_from_file_var` command from the exodus restart file defined in the `Mesh` block.
+
+!listing msr/msfr/thermochemistry/thermo.i block=AuxVariables
+
+Currently, the implementation of Thermochimica in MOOSE computes nodal values, so these finite volume variables must be converted to nodal variables which are the `tfuel_nod` and `pressure_nod` variables respectively. These variables are then used in the `ChemicalComposition` block.
+
+#### Auxiliary Kernels
+
+Correspondingly, the `AuxiliaryKernels` block sets up functors which will operate on the auxiliary variables. In this case, since a conversion from the finite volume variables to nodal variables is required, the `ProjectionAux` is used to perform this transformation. This is executed on `Initial` so that the nodal version of the variables are ready for the Thermochimica calculation.
+
+!listing msr/msfr/thermochemistry/thermo.i block=AuxKernels
+
+#### Initial Conditions
+
+Next, the `Initial Conditions` block sets up the initial conditions for various Auxiliary variables. In this case, the element variables are set through a `FunctionIC` which calculates a homogenous molar distribution from a depletion calculation and will be discussed shortly.
+
+!listing msr/msfr/thermochemistry/thermo.i block=ICs
+
+#### Problem
+
+The `Problem` block is used here to tell MOOSE that a finite element solve is not required, since a MOOSE-wrapped application via `UserObject` is being used instead. Additionally, the `allow_initial_conditions_with_restart` argument is used to allow for reading in the temperature and pressure variables from the multiphysics restart file.
```

Comment:
```suggestion
The `Problem` block is used here to tell MOOSE that a finite element solve is not required, since a MOOSE-wrapped application via `UserObject` is being used instead. Additionally, the `allow_initial_conditions_with_restart` argument is used to allow for reading in the temperature and pressure variables from the multiphysics restart file, while initializing other variables from the input file.
```

Reply:
Accepted into new push.